### PR TITLE
Fix `clippy` warnings and bump `srp` and `aucpace` MSRV to `1.61`

### PR DESCRIPTION
**Pull request by brxken128**
While playing around in the workspace (`SPAKE2` specifically), I noticed that there were a tonne of clippy warnings so I decided to fix them all.

~~My only concern is the silencing of `write` via `#[allow(clippy::unused_io_amount)]` - I wonder if changing all of those writes to `write_all` would produce more-than-negligible performance losses.~~ These were changed to `write_all` for correctness.

This changes the MSRV of `srp` and `aucpace` from `1.60.0` to `1.61.0` otherwise they won't build with some `const fn`s ([#93706](https://github.com/rust-lang/rust/issues/93706) was added in `1.61.0`)

### CODE REVIEW

**Comment by tarcieri** [comment]
cc @tritoke

**Comment by tritoke** [comment]
Thanks for fixing these, I got quite busy writing my dissertation towards the end of finishing this so it's nice that it's been tidied up!

All of the `.write` calls should be `.write_all`, they are only in examples so it's never failed for me in testing but it's always worth using the right method :)

**Comment by newpavlov** [comment]
It could be worth to also bump Clippy version in our [CI job](https://github.com/RustCrypto/PAKEs/blob/master/.github/workflows/workspace.yml#L19) and if there are warnings in other crates, it would be nice to fix them as well.

**Comment by brxken128** [comment]
Will draft this temporarily and get all of that done!

**Comment by tarcieri** [comment]
@brxken128 you can bump `aucpace` MSRV if need be

**Comment by brxken128** [comment]
> @brxken128 you can bump `aucpace` MSRV if need be

I think I broke builds anyway, probably removing the wrong `pub(crate)`. Shouldn't be too hard to fix.

I'll see what the new MSRV should be, from a glance 1.65.0 looks suitable but I'm not 100%.

**Comment by brxken128** [comment]
After further testing, `1.61.0` is all that's needed for things to function - both `srp` and `aucpace` would need these changes though.

**Comment by tritoke** [comment]
@brxken128 fwiw there is a tool called cargo-msrv that uses a kind of git bisect style approach to find the MSRV and it means you don't need to install all the toolchains, have found it very useful in the past so I thought I'd mention it :)

**Comment by brxken128** [comment]
> @brxken128 fwiw there is a tool called cargo-msrv that uses a kind of git bisect style approach to find the MSRV and it means you don't need to install all the toolchains, have found it very useful in the past so I thought I'd mention it :)

Oh wow that does sound incredibly helpful, I'll add it to my toolkit. I have a taskfile which installs all of the helpful tooling and this'll definitely be included - thank you so much!

**Comment by tarcieri** [comment]
> both `srp` and `aucpace` would need these changes though.

That's fine... can you bump them to the next minor version and add `-pre` though?

File: aucpace/Cargo.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ categories = ["cryptography", "authentication"]
 exclude = [".gitignore"]
 readme = "README.md"
 edition = "2021"
-rust-version = "1.60"
+rust-version = "1.61.0"
```
Comment (newpavlov) [inline aucpace/Cargo.toml:15]: It's fine to use `rust-version = "1.61"`, same for README.

File: aucpace/README.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ USE AT YOUR OWN RISK!
 
 ## Minimum Supported Rust Version
 
-Rust **1.60** or higher.
+Rust **1.61.0** or higher.
```
Comment (tarcieri) [inline aucpace/README.md:59]: Likewise you can just call this `1.61`

**Comment by brxken128** [comment]
> > both `srp` and `aucpace` would need these changes though.
> 
> That's fine... can you bump them to the next minor version in Cargo.toml and add `-pre` though? (i.e. v0.2.0-pre for `aucpace`, v0.7.0-pre for `srp`)

Done, and changed all `1.61.0` to `1.61`

**Pull request merged**